go/src/koding/kites/kloud/stack/provider/builder.go

```diff
@@ -398,43 +334,63 @@ func (b *Builder) MachineUIDs() map[string]string {
 // BuildCredentials fetches credential details for current b.Stack from MongoDB.
 //
 // When nil error is returned, the b.Koding and  b.Credentials fields are non-nil.
+//
+// TODO(rjeczalik): Replace with *credential.Client
 func (b *Builder) BuildCredentials(method, username, groupname string, identifiers []string) error {
 	// fetch jaccount from username
 	account, err := modelhelper.GetAccount(username)
 	if err != nil {
-		return ResError(err, "jAccount")
-	}
-
-	// fetch jGroup from group slug name
-	group, err := modelhelper.GetGroup(groupname)
-	if err != nil {
-		return ResError(err, "jGroup")
+		return models.ResError(err, "jAccount")
 	}
 
 	// fetch jUser from username
 	user, err := modelhelper.GetUser(username)
 	if err != nil {
-		return ResError(err, "jUser")
+		return models.ResError(err, "jUser")
 	}
 
-	// validate if username belongs to groupnam
-	selector := modelhelper.Selector{
-		"targetId": account.Id,
-		"sourceId": group.Id,
-		"as": bson.M{
-			"$in": []string{"member"},
-		},
+	kodingMeta := &KodingMeta{
+		Email:     user.Email,
+		Username:  user.Name,
+		Nickname:  account.Profile.Nickname,
+		Firstname: account.Profile.FirstName,
+		Lastname:  account.Profile.LastName,
+		Hash:      account.Profile.Hash,
 	}
 
-	count, err := modelhelper.RelationshipCount(selector)
-	if err != nil || count == 0 {
-		return fmt.Errorf("username '%s' does not belong to group '%s'", username, groupname)
+	groupIDs := []bson.ObjectId{account.Id}
+
+	if groupname != "" {
+		// fetch jGroup from group slug name
```

Comment:
This is a leftover from previous implementation, this method is going to be replaced with credential.Client introduced with this PR once it's feature-complete.

https://github.com/koding/koding/blob/990b6eff97f0de6a867a42147d2af0ce44df36de/go/src/koding/kites/kloud/stack/provider/builder.go#L338